#159987687 Users should receive a JWT on Successful Registration

CEOehis opened the pull request:
#### What does this PR do?
enable users to receive a JWT on Successful Registration

#### Description of Task to be completed?
- Implement JWT helper
- Implement authentication/authorization middleware for protected resources
- Include relevant tests

#### How should this be manually tested?
Run 'npm install'
create a .env using the .env.sample template
Run 'npm run migrate'
Run 'npm run start:dev'
use postman to test `/api/users` and `/api/users/login` endpoints

#### Any background context you want to provide?
The app previously used sessions with passport.js strategy for authentication.

#### What are the relevant pivotal tracker stories?
[#159987687](https://www.pivotaltracker.com/story/show/159987687)

CEOehis committed:
ft(jwt): add token utility

ft(jwt): add authorization middleware

Authorization middleware will be used instead of passport
to perform authorization for protected routes.
- Added relevant tests

CEOehis committed:
ft(jwt): return jwt with user signup data

- Remove hash from returned user data
- send generated token with data
- extend relevant test

ft(jwt): return token with user data on login

CEOehis committed:
ft(jwt): enable protected resources with jwt

The protected routes(resources) require a form of authorisation
in order to access them. This change uses an authorisation middleware
to limit access only to request that include a valid toke in the
authorization Header.
- update relevant tests
- protect routes with auth middleware

CEOehis committed:
ft(tests) add unit test for user model

d-beloved reviewed (approved):
The PR looks ok to me safe the errors raised by the other reviewer

lauragift21 reviewed (approved):
LGTM, But you didn't specify the user role in your JWT payload that is something you can add too.

CEOehis committed:
feat(jwt): encode user role in jwt payload

CEOehis committed:
feat(jwt): add middleware methods for validating priviledged roles

- Added authorizeAuthor() method for authorizing authors
- Added authorizeAdmin() method for authorizing admins
- Include relevant tests
- Add user model seeder

aknwosu commented:
LGTM
